tests/disaster-recovery/mdr/test_app_failover_and_relocate.py

```diff
@@ -0,0 +1,226 @@
+
+logger = logging.getLogger(__name__)
+
+polarion_id_primary_up = ""
+polarion_id_primary_down = "OCS-4346"
+
+
+@tier1
+class TestApplicationFailoverAndRelocate:
+    """
+    Test Failover and Relocate actions for a busybox application
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request, dr_workload):
+        """
+        If fenced, unfence the cluster and reboot nodes
+        """
+
+        def finalizer():
+            if self.drcluster_name and get_fence_state(self.drcluster_name) == "Fenced":
+                enable_unfence(self.drcluster_name)
+                gracefully_reboot_ocp_nodes(self.namespace, self.drcluster_name)
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.parametrize(
+        argnames=["primary_cluster_down"],
+        argvalues=[
+            pytest.param(
+                False,
+                marks=pytest.mark.polarion_id(polarion_id_primary_up),
+                id="primary_up",
+            ),
+            pytest.param(
+                True,
+                marks=pytest.mark.polarion_id(polarion_id_primary_down),
+                id="primary_down",
+            ),
+        ],
+    )
+    def test_application_failover_and_relocate(
+        self,
+        setup_acm_ui,
+        primary_cluster_down,
+        nodes_multicluster,
+        dr_workload,
+        node_restart_teardown,
+    ):
+        """
+        Tests to verify application failover and relocate between managed clusters
+        There are two test cases:
+            1) Failover and relocate of only application between managed clusters
+            2) Failover to secondary cluster when primary cluster is DOWN and Relocate
+                back to primary cluster once it recovers
+
+        This test is also compatible to be run from UI,
+        pass the yaml conf/ocsci/dr_ui.yaml to trigger it.
+        """
+
+        if config.RUN.get("mdr_failover_via_ui"):
+            ocs_version = version.get_semantic_ocs_version_from_config()
+            if ocs_version <= version.VERSION_4_12:
+                logger.error(
+                    "ODF/ACM version isn't supported for Failover/Relocate operation"
+                )
+                raise NotImplementedError
+
+        acm_obj = AcmAddClusters()
+        workload = dr_workload(num_of_subscription=1)[0]
+
+        # Create application on Primary managed cluster
+        set_current_primary_cluster_context(workload.workload_namespace)
+        primary_cluster_index = config.cur_index
+        node_objs = get_node_objs()
+        primary_cluster_name = get_current_primary_cluster_name(
+            namespace=workload.workload_namespace
+        )
+        self.drcluster_name = primary_cluster_name
+        self.namespace = workload.workload_namespace
+
+        # Stop primary cluster nodes
+        if primary_cluster_down:
+            logger.info("Stopping primary cluster nodes")
+            nodes_multicluster[primary_cluster_index].stop_nodes(node_objs)
```

Comment:
After nodes are down, don't we want to check the cluster status on ACM console using function `check_cluster_status_on_acm_console` in case Failover is done via UI?

Reply:
Pending

Reply:
this been added below so marking them resolved

Reply:
now looks fine

Reply:
The same check `check_cluster_status_on_acm_console` is missing when failover is performed by UI but primary cluster is up and running. @Akarsha-rai

Reply:
added